src/main/java/org/myrobotlab/programab/handlers/oob/OobProcessor.java

```diff
@@ -0,0 +1,93 @@
+package org.myrobotlab.programab.handlers.oob;
+
+import java.util.List;
+
+import org.myrobotlab.codec.CodecUtils;
+import org.myrobotlab.framework.Message;
+import org.myrobotlab.programab.models.Mrl;
+import org.myrobotlab.programab.models.Oob;
+import org.myrobotlab.service.ProgramAB;
+
+public class OobProcessor {
+
+  private static OobProcessor instance;
+  private transient ProgramAB programab;
+  protected int maxBlockTime = 2000;
+
+  private OobProcessor() {
+  }
+
+  public static OobProcessor getInstance(ProgramAB programab) {
+    if (instance == null) {
+      instance = new OobProcessor();
+      instance.programab = programab;
+    }
+    return instance;
+  }
+
+  public Message toMsg(Mrl mrl) {
+    Object[] data = null;
+    if (mrl.params != null) {
+      data = new Object[mrl.params.size()];
+      for (int i = 0; i < data.length; ++i) {
+        data[i] = mrl.params.get(i).trim();
+      }
+    }
+    String service = mrl.service == null?null:mrl.service.trim();
+    return Message.createMessage(programab.getName(), service, mrl.method.trim(), data);
+  }
+
+  public String process(Oob oob, boolean block) {
+    StringBuilder sb = new StringBuilder();
+
+    // FIXME dynamic way of registering oobs
+    if (oob != null) {
+      // Process <oob><mrl>
+      if (oob.mrl != null) {
+        List<Mrl> mrls = oob.mrl;
+        for (Mrl mrl : mrls) {
+          if (!block) {
+            // programab.out(toMsg(mrl));
+            programab.info("sending without blocking %s", toMsg(mrl));
+            programab.send(toMsg(mrl));
+          } else {
+            try {
+              programab.info("sendingBlocking without blocking %s", toMsg(mrl));
+              Object o = programab.sendBlocking(toMsg(mrl), maxBlockTime);
+              if (o != null) {
+                sb.append(o);
+              }
+            } catch (Exception e) {
+              programab.error(e);
+            }
+          }
+        }
+      } // for each mrl
+    }
+
+    // Process <oob><mrljson>
+    if (oob != null && oob.mrljson != null) {
+
+      Message[] msgs = CodecUtils.fromJson(oob.mrljson, Message[].class);
```

Comment:
Doesn't go through the proper codec utils message deserialization, so things like method cache isn't going to work. Side note: we should just move the specialized logic into a custom message deserializer now that we don't need to support GSON, but that's a project for another day

Reply:
This isn't invoking a single message .. this is unloading all the json messages .. so mrljson can have a set of msgs
[ { "name":"runtime", "method":"getUptime" ... },
{ "name":"runtime", "method":"getUptime" ... },
{ "name":"runtime", "method":"getUptime" ... },
... etc ]

Reply:
Yeah, but the issue is there's special logic to deserialize a single message, and that isn't used when going through the normal from json method, so none of the messages in the array will be deserialized properly